pkg/operator/controller/externaldns/deployment_test.go

```diff
@@ -1150,6 +1316,68 @@ func TestDesiredExternalDNSDeployment(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:             "Private Zone Azure",
```

Comment:
Does it make sense to repeat the same test for `Route` source? Will this go via different code path?

Reply:
I think this can be removed as we already have tested the code in the `Service` section.